crates/ruma-common/src/events/room/message/sanitize/html_sanitizer.rs

```diff
@@ -0,0 +1,547 @@
+    /// children will still be present in the output.
+    ///
+    /// To remove all tags, set this to `Some(Vec::new())`.
+    pub allowed_tags: Option<BTreeSet<&'a str>>,
+
+    /// HTML tags whose content will be removed.
+    ///
+    /// These tags will be removed from the output with their children.
+    ///
+    /// If a tag is both in `allowed_tags` and `remove_content_tags`, the behavior is unexpected.
+    pub remove_content_tags: BTreeSet<&'a str>,
+
+    /// HTML attributes per tag that will be left in the output of the sanitizer.
+    ///
+    /// If this is `None`, all the attributes are allowed on all tags.
+    ///
+    /// If this is `Some`, attributes that are not present in this list will be removed, even on
+    /// tags that are not listed.
+    ///
+    /// To remove all attributes, set this to `Some(BTreeMap::new())`.
+    pub allowed_attributes: Option<BTreeMap<&'a str, BTreeSet<&'a str>>>,
+
+    /// URI scheme per tag and attribute tuple that will be left in the output of the sanitizer.
+    ///
+    /// If this is `None`, all the URIs are allowed.
+    ///
+    /// If this is `Some`, URIs that are not present in this list will be removed. Tags and
+    /// attributes tuples that are not in this list are ignored.
+    ///
+    /// If no attribute that allows a URI is allowed, this will have no effect.
+    pub allowed_schemes: Option<BTreeMap<(&'a str, &'a str), BTreeSet<&'a str>>>,
+
+    /// Class name per tag that will be left in the output of the sanitizer.
+    ///
+    /// The class names to match allow the following wildcards:
+    ///
+    /// * `?` matches exactly one occurrence of any character.
+    /// * `*` matches arbitrary many (including zero) occurrences of any character.
+    ///
+    /// If this is `None`, all the class names are allowed.
+    ///
+    /// If this is `Some`, class names that are not present in this list will be removed. Tags that
+    /// are not in this list are ignored.
+    ///
+    /// If no `class` attribute is allowed, this will have no effect.
+    pub allowed_classes: Option<BTreeMap<&'a str, BTreeSet<&'a str>>>,
+
+    /// The maximum depth at which tags can be nested.
+    ///
+    /// If this is `None`, any depth is allowed.
+    ///
+    /// If this is `Some`, all tags deeper than this will be removed.
+    pub max_depth: Option<u32>,
+}
+
+impl<'a> HtmlSanitizer<'a> {
+    /// Constructs a `HTMLSanitizer` configured with the strict lists.
+    ///
+    /// It can also optionally remove the [rich reply fallback].
+    ///
+    /// [rich reply fallback]: https://spec.matrix.org/v1.2/client-server-api/#fallbacks-for-rich-replies
+    pub fn new(remove_reply_fallback: RemoveReplyFallback) -> Self {
+        let (allowed_tags, remove_content_tags) =
+            if remove_reply_fallback == RemoveReplyFallback::Yes {
+                let allowed_tags = BTreeSet::from(ALLOWED_TAGS_STRICT_WITHOUT_REPLY);
```

Comment:
Looks like we may want to use [phf](https://docs.rs/phf/latest/phf/) for these constants.

Reply:
Looked into phf. It's kind of annoying that we can't use the macros to make a map of sets.
I'd have to declare the sets outside the map, like that:

```rust

static INNER1: Set<&str> = phf_set! { "foo", "bar" };
static INNER2: Set<&str> = phf_set! { "baz", "bar" };

static MY_MAP: phf::Map<&str, Set<&str>> = phf_map! {
    "hello" => INNER1,
    "world" => INNER2,
};
```
If you're okay with that, I can do it.

Reply:
That sound reasonable.